### PR TITLE
add timestamp specific `GENERAL` configs

### LINKED ISSUE #284: `ProjectParameters_GENERAL.yml` cannot work for all timestamps
actually, this is because we need to update the scenarios here
https://github.com/RMI-PACTA/workflow.transition.monitor/blob/b7ed31ade67bceadb9d86858b651b078fed53ed3/parameter_files/ProjectParameters_GENERAL.yml#L57-L61

which then presents the problem that we cannot have a `ProjectParameters_GENERAL.yml` that works for all GENERAL projects for all timestamps 🤔

_Originally posted by @cjyetman in https://github.com/RMI-PACTA/workflow.transition.monitor/issues/283#issuecomment-2009243997_

Because the `ProjectParameters_GENERAL.yml` config (determined by `paste0("ProjectParameters_", project_code, ".yml")`) sets what scenarios should be used, any timestamps using GENERAL must use the same scenarios (and other configured parameters), which is a problem if the scenarios need to be different, e.g. between GENERAL 2022Q4 and GENERAL 2023Q4. This implies that we should maintain separate config files for different combinations of GENERAL and various timestamps, much like we do for the COP initiatives. That also implies that instead of using `GENERAL` for the project/initiative code, we should start using `GENERAL_2022Q4` and `GENERAL_2023Q4` etc., which may have consequences for how Constructiva passes the appropriate project code @AlexAxthelm?

### PR DESCRIPTION
**Pull request by cjyetman**
close #284
supercedes #285 

This PR does two things:

1. replaces `parameter_files/ProjectParameters_GENERAL.yml` with a unique config file for each timestamp used with `GENERAL`, in the form `parameter_files/ProjectParameters_GENERAL_{TIMESTAMP}.yml`
2. adds code to each web_tool_script to choose the proper config filename, for `GENERAL` only, by combining the initiative (`project_code`) and the timestamp (`port_holdings_date`) (targeting the new unique config files added here)

The new config files are based on:
- `2019Q4` and `2020Q4`: based on the config file in the `2dii_pacta:0.3.5` Docker file currently used on PROD on TM for those timestamps
- `2021Q4`: based on the config file in the `transitionmonitordockerregistry.azurecr.io/rmi_pacta:2021q4_1.0.0` Docker file currently used on PROD on TM for that timestamp
- `2022Q4` and `2023Q4`: based on the config file [ProjectParameters_PA2024CH.yml](https://github.com/RMI-PACTA/workflow.transition.monitor/blob/main/parameter_files/ProjectParameters_PA2024CH.yml) currently being used for `PA2022CH` (but appropriately modified for the timestamp)

### CODE REVIEW

**Comment by jdhoffa** [comment]
Note to self to discuss that with Nayra

**Review by jdhoffa — APPROVED** [review]
lgtm! A nice elegant solution

File: web_tool_script_1.R [hunk 1]
```diff
@@ -25,7 +25,13 @@ set_webtool_paths(portfolio_root_dir)
 
 set_portfolio_parameters(file_path = file.path(par_file_path, paste0(portfolio_name_ref_all, "_PortfolioParameters.yml")))
 
-set_project_parameters(file.path(working_location, "parameter_files", paste0("ProjectParameters_", project_code, ".yml")))
+proj_param_filename <- ifelse(
```
Comment (AlexAxthelm) [inline web_tool_script_1.R:28]:
```suggestion
proj_param_filename <- dplyr::if_else(
```

File: web_tool_script_2.R [hunk 2]
```diff
@@ -25,7 +25,13 @@ set_webtool_paths(portfolio_root_dir)
 
 set_portfolio_parameters(file_path = file.path(par_file_path, paste0(portfolio_name_ref_all, "_PortfolioParameters.yml")))
 
-set_project_parameters(file.path(working_location, "parameter_files", paste0("ProjectParameters_", project_code, ".yml")))
+proj_param_filename <- ifelse(
```
Comment (AlexAxthelm) [inline web_tool_script_2.R:28]:
```suggestion
proj_param_filename <- dplyr::if_else(
```

File: web_tool_script_3.R [hunk 3]
```diff
@@ -24,7 +24,13 @@ set_webtool_paths(portfolio_root_dir)
 
 set_portfolio_parameters(file_path = file.path(par_file_path, paste0(portfolio_name_ref_all, "_PortfolioParameters.yml")))
 
-set_project_parameters(file.path(working_location, "parameter_files", paste0("ProjectParameters_", project_code, ".yml")))
+proj_param_filename <- ifelse(
```
Comment (AlexAxthelm) [inline web_tool_script_3.R:27]:
```suggestion
proj_param_filename <- dplyr::if_else(
```

**Review by AlexAxthelm — APPROVED** [review]

**Pull request merged**